### PR TITLE
Add `job_variables` support to triggers

### PR DESCRIPTION
**Pull request by urimandujano**
This PR allows users to add `job_variables` to their trigger definitions supplied via the CLI directly, loaded as a JSON file, loaded from a YAML file, and a `prefect.yaml` file. Unless the feature flag is enabled, we'll support defining `job_variables` on the trigger but the field _will be_ nulled out when we actually create the automation. 

Note: This PR will depend on currently in-flight work to add `job_variable` support to our automations service and will not be merged until those changes are released.

### Example
Passing job_variables directly on the CLI
```
prefect deploy ./flows/hello_world.py:hello -n hello-deployment 
  --trigger '{
    "enabled": true,
    "match": {
      "prefect.resource.id": "prefect.flow-run.*"
    }, 
   "job_variables": {
      "one": 1
   }
}'
```
### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]
I believe this also lets you specify the `job_variables` in the  deployments in your `prefect.yaml` too, right?

**Comment by serinamarie** [comment]
> Note: This PR will depend on currently in-flight work to add job_variable support to our automations service and will not be merged until those changes are released.

It might be a good idea to add the `do not merge` label here

**Review by chrisguidry — APPROVED** [review]
Looks good!

File: tests/test_deployments.py [hunk 1]
```diff
@@ -946,6 +964,74 @@ async def test_deployment_apply_syncs_triggers(
...
+        created_deployment_id = str(uuid4())
+
+        updates = {
+            PREFECT_API_URL: f"https://api.prefect.cloud/api/accounts/{uuid4()}/workspaces/{uuid4()}",
+            PREFECT_CLOUD_API_URL: "https://api.prefect.cloud/api/",
+        }
+        if enable_flag:
+            updates[PREFECT_EXPERIMENTAL_ENABLE_FLOW_RUN_INFRA_OVERRIDES] = True
+
+        with temporary_settings(updates=updates):
```
Comment (chrisguidry) [inline tests/test_deployments.py:994]: Let me know how this works out, I've had some trouble using `with temporary_settings` inside of `async` functions.  If you hit any snags with this, you'll need to pull them up to a sync fixture and `yield` out of it

Comment (abrookins) [reply tests/test_deployments.py:994]: Interesting. I haven't hit that yet, but I'll keep an eye out. Is the failure recognizable or does it show up as a timeout, etc.?

Comment (chrisguidry) [reply tests/test_deployments.py:994]: I think it shows up as logging noise about `ContextVar`s?

**Pull request merged**